### PR TITLE
taking special notation into account

### PR DESCRIPTION
**Pull request by elina-holmes**
the idea is to take words with special notation (e.g., *u) into account when we count words per sample but not when we want to find the most frequent words.

### CODE REVIEW

File: authorship/Function_file.py [hunk 1]
```diff
@@ -134,9 +134,12 @@ def compile_data_CGN(string):
 
 def get_frequent_words(speakers, n):
```
Comment (wowtor) [inline authorship/Function_file.py:135]: maybe add some docstr here to describe what this function does? Do you know how to use them?

File: authorship/data.py [hunk 2]
```diff
@@ -60,25 +60,24 @@ def store_data(path, speakers):
         f.write(json.dumps(speakers))
 
 
-all_words = 0
-
 def read_session(lines):
-    global all_words
-    global test
     speakers = []
     test = lines.read()
```
Comment (wowtor) [inline authorship/data.py]: test?

File: authorship/data.py [hunk 3]
```diff
@@ -60,25 +60,24 @@ def store_data(path, speakers):
         f.write(json.dumps(speakers))
 
 
-all_words = 0
-
 def read_session(lines):
```
Comment (wowtor) [inline authorship/data.py:63]: add docstr?

File: authorship/data.py [hunk 4]
```diff
@@ -60,25 +60,24 @@ def store_data(path, speakers):
...
+    # *v: non-Dutch  words,  *n:  new  non-existing  words,  *s:  street  words,
+    # *a: incomplete  words, *u:  distorted  words, *x: unclear word
+    # we keep the words with their notation (for *n, *a, *u, and *x) to be able to exclude them when we
+    # derive the set of the most frequent words
+    test = test.replace('start\tend\ttext\n', '').replace('.', '').replace('-', ' ').replace('?', '').replace('\n', ' ') \
+        .replace('xxx', '').replace('ggg', '').replace('vvv', '').replace('*v','').replace('*s','')
+    s = test.translate({ord(c): None for c in string.punctuation if c != '*'})
+    tk = WhitespaceTokenizer()
+    conv = tk.tokenize(s)
+    if len(conv) > 30:
```
Comment (wowtor) [inline authorship/data.py]: this is a constraint on the input of this function but not visible to a user. I think this should be either a parameter (preferred) or documented

**Review by wowtor — APPROVED** [review]

File: authorship/Function_file.py (hunk 1 above)
Comment (elina-holmes) [reply authorship/Function_file.py:135]: done

File: authorship/data.py (hunk 2 above)
Comment (elina-holmes) [reply authorship/data.py]: not sure why it was global or given this name, i changed it to lines_to_words

File: authorship/data.py (hunk 3 above)
Comment (elina-holmes) [reply authorship/data.py:63]: done

File: authorship/data.py (hunk 4 above)
Comment (elina-holmes) [reply authorship/data.py]: thanks for it. i removed it (it was a temp thing to check sth.)

**Pull request merged**